desc/integrals/_bounce_utils.py

```diff
@@ -710,46 +711,16 @@ def _get_extrema(knots, g, dg_dz, sentinel=jnp.nan):
     return ext, g_ext
 
 
-def _where_for_argmin(points, ext, g_ext, upper_sentinel):
-    z1, z2 = points
-    assert z1.ndim > 1 and z2.ndim > 1
-    # Given
-    #      z1 and z2 with shape (..., num pitch, num well)
-    # and ext, g_ext with shape (..., num extrema),
-    # add dims to broadcast
-    #      z1 and z2 with shape (..., num pitch, num well, 1).
-    # and ext, g_ext with shape (...,         1,        1, num extrema).
-    return jnp.where(
-        (z1[..., jnp.newaxis] < ext[..., jnp.newaxis, jnp.newaxis, :])
-        & (ext[..., jnp.newaxis, jnp.newaxis, :] < z2[..., jnp.newaxis]),
-        g_ext[..., jnp.newaxis, jnp.newaxis, :],
-        upper_sentinel,
-    )
+# We can use the non-differentiable argmin because we actually want the gradients
+# to accumulate through only the minimum since we are differentiating how our
+# physics objective changes wrt equilibrium perturbations not wrt which of the
+# extrema get interpolated to.
 
 
-def _where_for_fft_argmin(points, ext, g_ext, upper_sentinel):
-    z1, z2 = points
-    assert z1.ndim >= 1 and z2.ndim >= 1
-    # Given
-    #      z1 and z2 with shape (..., num well)
-    # and ext, g_ext with shape (..., num extrema),
-    # add dims to broadcast
-    #      z1 and z2 with shape (..., num well, 1).
-    # and ext, g_ext with shape (...,        1, num extrema).
-    return jnp.where(
-        (z1[..., jnp.newaxis] < ext[..., jnp.newaxis, :])
-        & (ext[..., jnp.newaxis, :] < z2[..., jnp.newaxis]),
-        g_ext[..., jnp.newaxis, :],
-        upper_sentinel,
-    )
-
-
-def interp_to_argmin(
-    h, points, knots, g, dg_dz, method="cubic", beta=-100, upper_sentinel=1e2
-):
+def interp_to_argmin(h, points, knots, g, dg_dz, method="cubic"):
```

Comment:
No suggestion here but can you explain what this function is doing?

Reply:
We interpolate the function `h` to the global minimum of `g` that lies between each bounce point. In our application $g = \vert B \vert$ and $h$ is either $\nabla \psi$ or $e_{\alpha}$ at fixed $\rho, \phi$.